src/events.ts

```diff
@@ -57,6 +57,9 @@ export type DeviceRemovedEvent = CommonDeviceEvent<"device.removed">
 
 export type DeviceDeletedEvent = CommonDeviceEvent<"device.deleted">
 
+export type DeviceThirdPartyIntegrationDetected =
+  CommonDeviceEvent<"device.third_party_integration_detected">
```

Comment:
The only thing I'd say is we should decide if we want to add a separate event for when we detect the 3pp integration is no longer present, or if we add a property to this event that's a boolean to indicate whether the 3pp integration was detected as present or we detected it no longer present. Guesty who requested the event wanted to have it so they could know when the warning was added/removed so they could update their UI.

Reply:
That make sense. How can we distinguish one from another? Do we have the data to make this distinction?

Reply:
We can introduce a separate event i think, device.third_party_integration_removed

Reply:
How can I tell if it was removed? the warning only checks for the string `operto` in the access code name rn.